src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -79,45 +73,75 @@ export class HumanDescriptionMapper {
     }
   }
 
-  createHumanDescription(
-    descriptionFragments: HumanDescriptionFragment[],
-    token: Token | null,
-  ): string {
-    return descriptionFragments
-      .map((fragment) => {
+  async enrichFragments(
+    fragments: HumanDescriptionFragment[],
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichHumanDescriptionFragment[]> {
+    return Promise.all(
+      fragments.map(async (fragment) => {
         switch (fragment.type) {
           case ValueType.TokenValue:
-            if (!token?.decimals) return fragment.value.amount;
-
-            // Unlimited approval
-            if (fragment.value.amount === MAX_UINT256) {
-              return `unlimited ${token.symbol}`;
-            }
-
-            return `${formatUnits(fragment.value.amount, token.decimals)} ${
-              token.symbol
-            }`;
+            return this.enrichTokenValue(fragment, transaction, chainId);
           case ValueType.Address:
-            return this.shortenAddress(fragment.value);
           default:
-            return fragment.value;
+            return fragment;
         }
-      })
-      .join(' ');
+      }),
+    );
   }
 
-  shortenAddress(address: Hex, length = 4): string {
-    if (!isAddress(address)) {
-      throw Error('Invalid address');
-    }
+  async enrichTokenValue(
+    fragment: TokenValueFragment,
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichTokenValueFragment> {
+    try {
+      const token = await this.tokenRepository.getToken({
+        chainId,
+        address: transaction.to,
+      });
```

Comment:
Here we can ignore the result of the token if we don't get any. Therefore token can be `Token | null`:

```typescript
const token: Token | null = await this.tokenRepository
  .getToken({
    chainId,
    address: transaction.to,
  })
  .catch(() => null);

// taken from my comment below 👇
let amountLabel: string;
if (fragment.value.amount === MAX_UINT256) {
  amountLabel = 'unlimited';
} else if (token && token.decimals) {
  amountLabel = formatUnits(fragment.value.amount, token.decimals);
} else {
  amountLabel = fragment.value.amount.toString();
}

return <RichTokenValueFragment>{
  type: ValueType.TokenValue,
  value: {
    token,
    amount: amountLabel,
  },
};
```

Reply:
This was my first approach as well but somehow the `catch` clause here is not catching an error when throwing in the test so I changed it to use `try catch` which works.